src/disc/stk/Albany_OrdinarySTKFieldContainer_Def.hpp

```diff
@@ -68,12 +67,31 @@ OrdinarySTKFieldContainer<Interleaved>::OrdinarySTKFieldContainer(
   typedef typename AbstractSTKFieldContainer::VectorFieldType       VFT;
   typedef typename AbstractSTKFieldContainer::ScalarFieldType       SFT;
 
-  int num_time_deriv = params_->get<int>("Number Of Time Derivatives");
+  num_params = num_params_; 
 
+  int num_time_deriv = params_->get<int>("Number Of Time Derivatives");
 #ifdef ALBANY_DTK
-  int output_dtk_field =
+  bool output_dtk_field =
       params_->get<bool>("Output DTK Field to Exodus", false);
 #endif
+  //IKT FIXME? - currently won't write dxdp to output file if problem is steady, 
+  //as this output doesn't work in same way.  May want to change in the future.
+  bool output_dxdp_field = false; 
+  if (num_params_ > 0 && num_time_deriv > 0) output_dxdp_field = true; 
+
```

Comment:
It would be nice to have some way to control the whether to output dxdp from input file.

Reply:
This would be very easy to do - it would be just like this code: https://github.com/SNLComputation/Albany/blob/observer-trans-sens/src/disc/stk/Albany_OrdinarySTKFieldContainer_Def.hpp#L74-L75 .  I actually thought the opposite of what you said, based on my experience with the DTK output (I put that Exodus output option in for DTK to turn it off from PL) - I thought that it's better not to give an additional PL option.  For the DTK output stuff, it becomes a huge mess b/c people forget that the option is in the input file then wonder with the DTK stuff's not there.  Right now, dxdp only get written for a transient problem with sensitivities enabled, so for the majority of Albany problems, it won't get written.  Anyway, I'm happy to add a PL option similar to DTK if you think it's better.  Would you like me to do that?

Reply:
OK, I see your point. It's just that sometime the mesh we get are really big when we are working on large problem. Anyway, since, as you said, it is stored only for transient problems with sensitivities enable, I think it make sense to live it as is. We can always change it in a second time. Thanks.

Reply:
I'll go ahead and resolve this point then.  It's very easy to add the option - just one like, similar to what's done for the DTK output.  We can always do it in the future, like you said.